comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,237 @@
+	for pid, procA := range procCacheA {
+		procB, exists := procCacheB[pid]
+
+		// new process
+		if !exists {
+			newProcs = append(newProcs, procA)
+		} else if procB.Stats.CreateTime != procA.Stats.CreateTime {
+			// same process PID exists, but different process due to creation time
+			newProcs = append(newProcs, procA)
+		}
+	}
+	return newProcs
+}
+
+// collect captures all the required process data for the process check
+func (c *collector) collect(ctx context.Context, collectionTicker *clock.Ticker) {
+	// TODO: implement the full collection logic for the process collector. Once collection is done, submit events.
+	ctx, cancel := context.WithCancel(ctx)
+	defer collectionTicker.Stop()
+	defer cancel()
+	for {
+		select {
+		case <-collectionTicker.C:
+			// Fetch process data and submit events to streaming channel for asynchronous processing
+			procs, err := c.processProbe.ProcessesByPID(time.Now(), false)
+			if err != nil {
+				log.Errorf("Error getting processes by pid: %v", err)
+				return
+			}
+
+			// categorize the processes into events for workloadmeta
+			createdProcs := processCacheDifference(procs, c.lastCollectedProcesses)
+			var wlmCreatedProcs []*workloadmeta.Process
+			for _, proc := range createdProcs {
+				wlmCreatedProcs = append(wlmCreatedProcs, processToWorkloadMetaProcess(proc))
+			}
+
+			deletedProcs := processCacheDifference(c.lastCollectedProcesses, procs)
+			var wlmDeletedProcs []*workloadmeta.Process
+			for _, proc := range deletedProcs {
+				wlmDeletedProcs = append(wlmDeletedProcs, processToWorkloadMetaProcess(proc))
+			}
+
+			// send these events to the channel
+			c.processEventsCh <- &Event{
+				Created: wlmCreatedProcs,
+				Deleted: wlmDeletedProcs,
+			}
+
+			// store latest collected processes
+			c.lastCollectedProcesses = procs
+		case <-ctx.Done():
+			log.Infof("The %s collector has stopped", collectorID)
+			return
+		}
+	}
+}
+
+// stream processes events sent from data collection and notifies WorkloadMeta that updates have occurred
+func (c *collector) stream(ctx context.Context) {
+	healthCheck := health.RegisterLiveness(componentName)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	for {
+		select {
+		case <-healthCheck.C:
+
+		case processEvent := <-c.processEventsCh:
+			var events []workloadmeta.CollectorEvent
+			for _, proc := range processEvent.Deleted {
+				events = append(events, workloadmeta.CollectorEvent{
+					Type:   workloadmeta.EventTypeUnset,
+					Entity: proc,
+					Source: workloadmeta.SourceProcessCollector,
+				})
+			}
+
+			for _, proc := range processEvent.Created {
+				events = append(events, workloadmeta.CollectorEvent{
+					Type:   workloadmeta.EventTypeSet,
+					Entity: proc,
+					Source: workloadmeta.SourceProcessCollector,
+				})
+			}
+
+			c.store.Notify(events)
+
+		case <-ctx.Done():
+			err := healthCheck.Deregister()
+			if err != nil {
+				log.Warnf("error de-registering health check: %s", err)
+			}
```

Comment:
Nit: should we also have an INFO log that the stream goroutine has stopped, similar to the collector?